plugins/cpp/parser/src/clangastvisitor.h

```diff
@@ -435,15 +443,12 @@ class ClangASTVisitor : public clang::RecursiveASTVisitor<ClangASTVisitor>
     cppRecord->entityHash = astNode->entityHash;
     cppRecord->name = rd_->getNameAsString();
     cppRecord->qualifiedName = rd_->getQualifiedNameAsString();
-    if (const clang::CXXRecordDecl* crd
-        = llvm::dyn_cast<clang::CXXRecordDecl>(rd_))
+
+    if (clang::CXXRecordDecl* crd = llvm::dyn_cast<clang::CXXRecordDecl>(rd_))
```

Comment:
Why are we losing the `const` here?

Reply:
We only really 'lost' the const qualifier for the first two lines (when assigning isPOD and isAbstract). As for the code below //--- CppInheritance ---//: crd was non-const there even before this commit, so no loss there.
What I essentially did here is that I noticed that two consecutive if statements had the exact same dyn_cast condition, so I put them inside one common if statement instead to reduce code duplication and the same dyn_cast overhead twice, and kept the condition without the const qualifier.
We can of course keep the const as well, I just did not want to 'restrict' the code to having to use const-only functionality on an object that we otherwise received as non-const in the function argument. In this particular case, the code seems to compile with crd being const as well, so I'll consider marking it const.